components/p3a/brave_p3a_service.cc

```diff
@@ -85,6 +96,32 @@ base::Time NextMonday(base::Time time) {
   return result;
 }
 
+base::Time NextDay(base::Time time) {
+  return (time.LocalMidnight() + base::Days(1) + base::Hours(4))
+      .LocalMidnight();
+}
+
+const char* GetRotationTimestampPref(MetricLogType log_type) {
+  switch (log_type) {
+    case MetricLogType::kTypical:
+      return kLastTypicalRotationTimeStampPref;
+    case MetricLogType::kExpress:
+      return kLastExpressRotationTimeStampPref;
+  }
+  NOTREACHED();
```

Comment:
Chromium uses the following and always returns a value, even if an empty string;
```
  NOTREACHED() << "Unexpected value for Value: " << value_;
  return kUndefinedType;
```